### PR TITLE
feat(aws): allow more customisation in prompt function

### LINKED ISSUE #11618: more detailed customization for `aws` plugin prompt
### If the feature request is for a plugin or theme, specify it here.

aws

### If the feature solves a problem you have, specify it here.

_No response_

### Describe the proposed feature.

1. Empty variable should be treated as null variable, not unset variable.

if I `export ZSH_THEME_AWS_PROFILE_PREFIX=""`, it expands to`<aws:` because of the following code:
```
${ZSH_THEME_AWS_PROFILE_PREFIX:=<aws:}
```

([here](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/aws/aws.plugin.zsh#L202) and [here](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/aws/aws.plugin.zsh#L206))

2. Space between profile and region should be option, not mandatory.

```
_aws_to_show+=" "
```

I think it makes sense but it would be greater if I can remove or override this. I want something like `[admin/us-west-1]`.

([here](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/aws/aws.plugin.zsh#L205))

### Describe alternatives you've considered

for feature 1 : use `-z` condition statement instead of `:=` expansion.

for feature 2 : make a space to optional (customizable) variable.

If moderators allow, I want to resolve this issue with my code.

### Additional context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by roeniss**
resolves #11618

tl;dr : `$(aws_prompt_info)` now can be `admin|us-east-1`

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes

- feature 1: allow empty string for variables
  - e.g., `ZSH_THEME_AWS_PROFILE_PREFIX` can be an empty string (`""`)
- feature 2: add a divider variable instead of a single space

## Other comments:

Slight worry for the second feature: Before this commit, the divider (a single space) i visible only when AWS_PROFILE and AWS_REGION were set. But this PR, for straightforward usage, makes it always visible.

### CODE REVIEW

File: plugins/aws/aws.plugin.zsh [hunk 1]
```diff
@@ -198,13 +198,32 @@ compctl -K _aws_profiles asp acp aws_change_access_key
 function aws_prompt_info() {
   local _aws_to_show
   local region="${AWS_REGION:-${AWS_DEFAULT_REGION:-$AWS_PROFILE_REGION}}"
+
   if [[ -n $AWS_PROFILE ]];then
-    _aws_to_show+="${ZSH_THEME_AWS_PROFILE_PREFIX:=<aws:}${AWS_PROFILE}${ZSH_THEME_AWS_PROFILE_SUFFIX:=>}"
```
Comment (carlosala) [inline plugins/aws/aws.plugin.zsh:202]: Just removing the `:` would do the trick. It should be `${ZSH_THEME_AWS_PROFILE_PREFIX=<aws:}`, for example.
There's no need of adding all those if cases.

File: plugins/aws/aws.plugin.zsh [hunk 2]
```diff
@@ -198,13 +198,32 @@ compctl -K _aws_profiles asp acp aws_change_access_key
...
+  fi
+
+  if [[ -z ${ZSH_THEME_AWS_DIVIDER+x} ]]; then
+    ZSH_THEME_AWS_DIVIDER=" "
   fi
+  _aws_to_show+="${ZSH_THEME_AWS_DIVIDER}"
+
   if [[ -n $AWS_REGION ]]; then
-    [[ -n $AWS_PROFILE ]] && _aws_to_show+=" "
-    _aws_to_show+="${ZSH_THEME_AWS_REGION_PREFIX:=<region:}${region}${ZSH_THEME_AWS_REGION_SUFFIX:=>}"
```
Comment (carlosala) [inline plugins/aws/aws.plugin.zsh:206]: Same happens here.

File: plugins/aws/aws.plugin.zsh [hunk 3]
```diff
@@ -198,13 +198,32 @@ compctl -K _aws_profiles asp acp aws_change_access_key
...
+    _aws_to_show+="${ZSH_THEME_AWS_PROFILE_PREFIX}${AWS_PROFILE}${ZSH_THEME_AWS_PROFILE_SUFFIX}"
+  fi
+
+  if [[ -z ${ZSH_THEME_AWS_DIVIDER+x} ]]; then
+    ZSH_THEME_AWS_DIVIDER=" "
   fi
+  _aws_to_show+="${ZSH_THEME_AWS_DIVIDER}"
+
   if [[ -n $AWS_REGION ]]; then
-    [[ -n $AWS_PROFILE ]] && _aws_to_show+=" "
```
Comment (carlosala) [inline plugins/aws/aws.plugin.zsh:205]: Let's also have this check. It'll prevent adding the divider if the region or profiles are not set.

File: plugins/aws/aws.plugin.zsh (hunk 1 above)
Comment (roeniss) [reply plugins/aws/aws.plugin.zsh:202]: I didn't know this trick. Thank you! It was so hard to find [the reference](https://tldp.org/LDP/abs/html/parameter-substitution.html)...  I'll change it.

File: plugins/aws/aws.plugin.zsh (hunk 2 above)
Comment (roeniss) [reply plugins/aws/aws.plugin.zsh:206]: https://github.com/ohmyzsh/ohmyzsh/pull/11619#discussion_r1171338084

File: plugins/aws/aws.plugin.zsh (hunk 3 above)
Comment (roeniss) [reply plugins/aws/aws.plugin.zsh:205]: My basic idea was that most users would have both profile and region env. But it seems not great guess so I will change it like this: 

```
_aws_to_show+="${ZSH_THEME_AWS_DIVIDER}" # moves this in the previous commit to the...

  if [[ -n $AWS_REGION ]]; then
    [[ -n $AWS_PROFILE ]] && _aws_to_show+="${ZSH_THEME_AWS_DIVIDER=' ' }" # here
```

What do you think? Let me just push a commit first so you can see it easily...

**Review by carlosala — APPROVED** [review]

**Comment by carlosala** [comment]
Thanks for your contribution!
